qutebrowser/browser/commands.py

```diff
@@ -643,14 +643,32 @@ def scroll_perc(self, perc: {'type': float}=None,
 
     @cmdutils.register(instance='command-dispatcher', hide=True,
                        scope='window', count='count')
-    def scroll_page(self, x: {'type': float}, y: {'type': float}, count=1):
+    def scroll_page(self, x: {'type': float}, y: {'type': float}, *,
+                    top_navigate: {'type': ('prev', 'decrement')}=None,
+                    bottom_navigate: {'type': ('next', 'increment')}=None,
+                    count=1):
         """Scroll the frame page-wise.
 
         Args:
             x: How many pages to scroll to the right.
             y: How many pages to scroll down.
+            bottom_navigate: :navigate to the next page on bottom
+            top_navigate: :navigate to the previous page on top
             count: multiplier
         """
+        frame = self._current_widget().page().currentFrame()
+        if not frame.url().isValid():  # Issue 701
+            return
```

Comment:
> Or will this always be the case for pages without any scrollbar?

[Yes](http://doc.qt.io/qt-5.4/qwebframe.html#scrollBarMaximum):

>  0 if no scrollbar is found for orientation.

I don't know if it's possible to have a page with an "invalid" URL, but it seems unlikly to me...

I don't know another way to check this. From what I remember, `toString()`
returns an empty string:

	>>> QUrl('http://127.0.0.1:1000')
	PyQt5.QtCore.QUrl('http://127.0.0.1:1000')

	>>> QUrl('http://127.0.0.1:not_valid')
	PyQt5.QtCore.QUrl('')

	>>> x = QUrl('http://127.0.0.:80801')
	>>> x
	PyQt5.QtCore.QUrl('http://127.0.0.1:8080')
	>>> x.setHost('!!!invalid!!!')
	>>> x
	PyQt5.QtCore.QUrl('')
	>>> x.port()
	8080